### PR TITLE
FederatedRyaExample

### PR DESCRIPTION
**Pull request by jasonyang1686**
Some example codes using federated Rya with current Rya state.

### CODE REVIEW

File: extras/indexingExample/src/main/java/FederatedRyaMockExample.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+import mvm.rya.indexing.accumulo.ConfigUtils;
+import mvm.rya.indexing.accumulo.geo.GeoConstants;
+import mvm.rya.indexing.external.tupleSet.PcjTables;
+import mvm.rya.indexing.external.tupleSet.PcjTables.PcjException;
+import mvm.rya.indexing.external.tupleSet.PcjTables.PcjVarOrderFactory;
+import mvm.rya.rdftriplestore.inference.InferenceEngineException;
+import mvm.rya.sail.config.RyaSailFactory;
+
+public class FederatedRyaMockExample {
+	private static final Logger log = Logger.getLogger(RyaDirectExample.class);
```
Comment (pujav65) [inline extras/indexingExample/src/main/java/FederatedRyaMockExample.java]: change the logging to reference the right class

File: extras/indexingExample/src/main/java/FederatedRyaRealExample.java [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+	// Connection configuration parameters
+	//
+
+	private static final boolean USE_MOCK_INSTANCE = false;
+	private static final boolean PRINT_QUERIES = true;
+	private static final String INSTANCE = "dev";
+	private static final String RYA_TABLE_PREFIX = "rya_";
+	private static final String AUTHS = "";
+	//the second VM's IP is 192.168.33.20
+	private static final String SESAME_SERVER="http://192.168.33.20:8080/openrdf-sesame";
```
Comment (pujav65) [inline extras/indexingExample/src/main/java/FederatedRyaRealExample.java]: don't check in ips

File: extras/indexingExample/src/main/java/FederatedRyaRealExample.java [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+		conf.set(ConfigUtils.USE_PCJ, "false");
+		conf.set(ConfigUtils.USE_GEO, "false");
+		conf.set(ConfigUtils.USE_FREETEXT, "false");
+		conf.set(ConfigUtils.USE_TEMPORAL, "false");
+		conf.set(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX,
+				RYA_TABLE_PREFIX);
+		conf.set(ConfigUtils.CLOUDBASE_USER, "root");
+		conf.set(ConfigUtils.CLOUDBASE_PASSWORD, "root");
+		conf.set(ConfigUtils.CLOUDBASE_INSTANCE, INSTANCE);
+		conf.set(ConfigUtils.CLOUDBASE_ZOOKEEPERS,"192.168.33.10:2181");
```
Comment (pujav65) [inline extras/indexingExample/src/main/java/FederatedRyaRealExample.java]: don't check in ips

**Comment by pujav65** [comment]
I'm not sure that we want to add all three of these examples.  There is a lot of duplicated code between them.  
I think that you likely would only want FederatedRyaMockExample_diff_prefix.java (though maybe rename that to FederatedRyaMockExample.java), and the sesame integration.  If you could combine the diff prefix and the sesame integration into the same example (so that they don't duplicate code anymore), I will merge this.

File: extras/indexingExample/src/main/java/FederatedRyaRealExample.java (hunk 2 above)
Comment (DLotts) [inline extras/indexingExample/src/main/java/FederatedRyaRealExample.java]: Note that this is a non-internet-routable address similar to the one used by our VM in the Rya Vagrant example.

**Pull request closed without merging**